release-24.2: clusterversion: set devBranch to false, version to beta.1

RaduBerinde opened the pull request:
**What this does:**

1. Sets `DevelopmentBranch = false`
2. Sets `version.txt` to `beta.1`

For REL-1048

**When to merge:**
This PR should not be merged until we intend to select the
`v24.2.0-beta.1` release candidate.

Release justification: necessary part of the release process.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/126167)
<!-- Reviewable:end -->

RaduBerinde committed:
release-24.2: clusterversion: set devBranch to false, version to beta.1

**What this does:**

1. Sets `DevelopmentBranch = false`
2. Sets `version.txt` to `beta.1`

For REL-1048

**When to merge:**
This PR should not be merged until we intend to select the
`v24.2.0-beta.1` release candidate.

Release justification: necessary part of the release process.

Release note: None
Epic: none
